### PR TITLE
Add tutorial to run object detection on your webcam

### PR DESCRIPTION
**Pull request by ThomasDelteil**
Based on a small demo that was built for ApacheCon.
See this [script](https://github.com/ThomasDelteil/ObjectDetectionWebcam_MXNet) for reference.

Since the script cannot run in the CI (no webcam), the code has been put in code blocks. The problem is that now the "Download the script" button is misleading. Happy to hear suggestion on how to improve the UX

### CODE REVIEW

**Comment by mli** [comment]
Job PR-361-1 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-361/1/index.html 
Code coverage of this PR:  vs. Master:

**Comment by hetong007** [comment]
@zhreshold do you have any comments?

@ThomasDelteil The check failed because of unstable CI. Can you please re-trigger it with an empty commit?

**Comment by mli** [comment]
Job PR-361-2 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-361/2/index.html 
Code coverage of this PR:  vs. Master:

**Comment by ThomasDelteil** [comment]
@zhreshold updated the code

**Comment by mli** [comment]
Job PR-361-3 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-361/3/index.html 
Code coverage of this PR:  vs. Master:

**Comment by zhreshold** [comment]
@ThomasDelteil Thanks for the awesome demo. I found it's not very convenient to download the code.

Using the built-in download button by sphinx-gallery will download all codes within code-block, which won't execute.  User has to copy & paste code blocks one by one manually. 

One way to overcome that is to put a command line friendly script file under `scripts/detection/demo_webcam.py` for example, and use directive
```
:download:`Download demo_webcam.py<../../../scripts/detection/demo_webcam.py>`
```
to allow user download it directly.

**Comment by ThomasDelteil** [comment]
@zhreshold, thanks sounds great, agreed it is not great UX at the moment. Will add the script and update the tutorial with the directive

**Comment by mli** [comment]
Job PR-361-4 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-361/4/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-361-7 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-361/7/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-361-9 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-361/9/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-361-10 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-361/10/index.html 
Code coverage of this PR:  vs. Master:

**Comment by ThomasDelteil** [comment]
@zhreshold update on that?

File: docs/tutorials/detection/demo_webcam.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
+"""8. Run an object detection model on your webcam
```
Comment (zhreshold) [inline docs/tutorials/detection/demo_webcam.py]: please increment to 9.

File: scripts/detection/demo_webcam_run.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+# Load the model
+net = gcv.model_zoo.get_model('ssd_512_mobilenet1.0_voc', pretrained=True)
+
+# Load the webcam handler
+cap = cv2.VideoCapture(0)
+time.sleep(1)  ### letting the camera autofocus
+
+axes = None
+NUM_FRAMES = 200  # you can change this
```
Comment (zhreshold) [inline scripts/detection/demo_webcam_run.py]: better use argparse? Also allow infinite frames?

**Comment by ThomasDelteil** [comment]
@zhreshold updated

**Comment by mli** [comment]
Job PR-361-17 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-361/17/index.html 
Code coverage of this PR:  vs. Master:

File: docs/tutorials/detection/demo_webcam.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+    import cv2
+    import gluoncv as gcv
+    import matplotlib.pyplot as plt
+    import mxnet as mx
+
+
+In this tutorial we use ``ssd_512_mobilenet1.0_voc``, a snappy network with good accuracy that should be
+well above 1 frame per second on most laptops. Feel free to try a different model from
+the :doc:`../../model_zoo/index` !
```
Comment (zhreshold) [inline docs/tutorials/detection/demo_webcam.py]: :doc:`..` does not get rendered properly. Try 
```
`Gluon Model Zoo <../../model_zoo/index.html>`__
```

Comment (ThomasDelteil) [reply docs/tutorials/detection/demo_webcam.py]: updated

**Comment by mli** [comment]
Job PR-361-18 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-361/18/index.html 
Code coverage of this PR:  vs. Master:

File: docs/tutorials/detection/demo_webcam.py (hunk 3 above)
Comment (zhreshold) [reply docs/tutorials/detection/demo_webcam.py]: Sorry, my bad, it's  ``../../model_zoo/detection`` actually

Comment (ThomasDelteil) [reply docs/tutorials/detection/demo_webcam.py]: hard to get right :)

**Comment by mli** [comment]
Job PR-361-19 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-361/19/index.html 
Code coverage of this PR:  vs. Master:

**Comment by zhreshold** [comment]
Cool, this is merged

**Pull request merged**